types/callMacros.d.ts

```diff
@@ -8,7 +8,7 @@ declare function assert<T>(condition: T, message?: string): asserts condition;
  * Returns the type of the given object as a string. This function works similarly to Lua’s native type function, with
  * the exceptions that Roblox-defined data types like Vector3 and CFrame return their respective data types as strings.
  */
-declare function typeOf(value: any): keyof CheckableTypes;
+declare function $typeof(value: any): keyof CheckableTypes;
```

Comment:
I don't think this is a positive change. Like `assert`, it's a macro which maps one-to-one to its Luau function. The $ prefix should indicate that the emit is significantly different or in some other way related to compiler behaviour. Everyone has already gotten used to the capitalisation `typeOf`, a breaking change here has no further value.